File: app/src/main/java/vn/loitp/app/activity/customviews/layout/scrollview2d/ScrollView2DActivity.java
package vn.loitp.app.activity.customviews.layout.scrollview2d;

import android.os.Bundle;

import com.core.base.BaseFontActivity;
import com.core.utilities.LLog;
import com.core.utilities.LUIUtil;
import com.views.scrollview.TwoDScrollView;

import loitp.basemaster.R;

public class ScrollView2DActivity extends BaseFontActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final TwoDScrollView twoDScrollView = findViewById(R.id.sv);
        twoDScrollView.setScrollChangeListner((view, x, y, oldx, oldy) -> {
            LLog.d(TAG, "setScrollChangeListner " + x + " - " + y);
        });
        LUIUtil.setDelay(2000, mls -> twoDScrollView.smoothScrollTo(300, 300));
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_scrollview_2d;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/layout/scrollview2d/ScrollView2DActivity.kt
package vn.loitp.app.activity.customviews.layout.scrollview2d

import android.os.Bundle
import androidx.core.view.isVisible
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import com.loitpcore.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_layout_scrollview_2d.*
import vn.loitp.app.R

@LogTag("ScrollView2DActivity")
@IsFullScreen(false)
class ScrollView2DActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_layout_scrollview_2d
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.viewShadow?.isVisible = true
            this.tvTitle?.text = ScrollView2DActivity::class.java.simpleName
        }
        twoDScrollView.setScrollChangeListener { _, x, y, oldX, oldY ->
            logD("setScrollChangeListener $x - $y - $oldX - $oldY")
        }
        LUIUtil.setDelay(
            mls = 2000,
            runnable = Runnable {
                twoDScrollView.smoothScrollTo(300, 300)
            }
        )
    }
}
